### PR TITLE
Optimize loading of data for satd kernels

### PR DESCRIPTION
**Pull request by shssoichiro**
Following the ASM update by redzic, this now only affects 4x16 and 16x4 transforms,
but since it is still in use, it is still worth pushing the more optimal assembly.

4x4 kernel Before: https://pastebin.com/GP4Vjb2z
4x4 kernel After: https://pastebin.com/Kr2nz7mZ

### CODE REVIEW

File: src/asm/x86/dist/hbd.rs [hunk 1]
```diff
@@ -61,9 +61,14 @@ macro_rules! satd_kernel_hbd_avx2 {
           let input1 = &mut Aligned::<[i32; 8 * 8]>::uninitialized().data[..size * size];
           let input2 = &mut Aligned::<[i32; 8 * 8]>::uninitialized().data[..size * size];
           for y in 0..(sizei) {
-            for x in 0..(sizei) {
-              input1.as_mut_ptr().offset(y * sizei + x).write(src.offset(y * src_stride + x).read() as i32);
-              input2.as_mut_ptr().offset(y * sizei + x).write(dst.offset(y * dst_stride + x).read() as i32);
+            if size == 8 {
```
Comment (lu-zero) [inline src/asm/x86/dist/hbd.rs:64]: I hope this branch is elided by the compiler.

**Comment by BlueSwordM** [comment]
I tested this patch, and for some reason, this periodically creates flat blocks within a frame, which is not normal behavior.

File: src/asm/x86/dist/hbd.rs (hunk 1 above)
Comment (shssoichiro) [reply src/asm/x86/dist/hbd.rs:64]: It is, I verified that the compiler sees `size` as constant and elides out any branches. So the 8x8 version has the first branch only, and the 4x4 version has the second branch only.

**Comment by shssoichiro** [comment]
Although apparently one of the tests is segfaulting... that's probably not good.

**Comment by shssoichiro** [comment]
Made obsolete by #3051

**Pull request closed without merging**